tox/config.py

```diff
@@ -258,11 +260,16 @@ def feedback(msg, sysexit=False):
         raise SystemExit(1)
 
 
-class VersionAction(argparse.Action):
-    def __call__(self, argparser, *args, **kwargs):
-        version = tox.__version__
-        print("%s imported from %s" % (version, tox.__file__))
-        raise SystemExit(0)
+def print_version_info(pm):
+    print("%s imported from %s" % (tox.__version__, tox.__file__))
+    plugin_dist_info = pm.list_plugin_distinfo()
+    if not plugin_dist_info:
+        return
+    print('registered plugins:')
+    for mod, egg_info in plugin_dist_info:
+        source = getattr(mod, '__file__', repr(mod))
```

Comment:
please take a look at trimming this down

site packages installed ones shouldn't list at all
develop installed ones should point to the project folder

Reply:
I don't understand what you mean. Also: AFAK it shows pretty much the same like `pytest --version`.

Reply:
then we should fix it for pytest, too

Reply:
I still don't understand what you mean, though, so I can't fix anything yet :) - if I do I will fix it here and open a PR on pytest also.

Reply:
we shouldnt print the site packages location for each plugin, instead its enough to know its in site pacakges usually.

only for the eggs it important to know where they where added

i do wonder if we should perhaps put the utility for listing plug-ins and their versions/locations into pluggy

Reply:
ah - now I get it - the paths. I actually do not have a problem with showing the real path to where it is imported from, just like for tox itself.

Reply:
I like the idea of having this the same over all tools that use pluggy, so I will open an issue there and gladly use that function of pluggy, when it is introduced.
 
For here and now, I merge this as it in as this is exactly parallel to what pytest outputs with the difference that pytest outputs to stderr (I don't know why) and tox outputs to stdout.
`pytest --version`:

    This is pytest version 3.2.2, imported from /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/pytest.py
    setuptools registered plugins:
      pytest-cov-2.5.1 at /home/oliver/.virtualenvs/tmp/lib/python3.6/site-packages/pytest_cov/plugin.py